### PR TITLE
Enable prefix bloom filter by default

### LINKED ISSUE #2601: [STO]Bloom filter (Currently we use full-key)

### PR DESCRIPTION
**Pull request by critical27**
1. Enable prefix bloom filter, close whole key bloom filter by default. (previously we use whole key bloom filter by default). Benchmark is listed [here](https://confluence.nebula-graph.io/pages/viewpage.action?pageId=10726186). In read/write scenario, the read performance is doubled, while pure read is not impacted. PS: the memory usage will shrink as well.
2. We need to get some property in rocksdb during test. So add a http interface to get rocksdb property, space name and legal property need to be provided. (No need to set `enable_rocksdb_statistics` to true). For example:

```
curl -G "http://127.0.0.1:52005/rocksdb_property?space=ldbc_snb_sf10&property=rocksdb.levelstats"
```

Close #2601

@randomJoe211, please check if we need to add a doc about the http interfaces.
@HarrisChu, I have check the compatibility during test, everything works fine. Please pay attention to compatible test later. (write new data to old storage, check data)

### CODE REVIEW

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -1168,5 +1168,26 @@ nebula::cpp2::ErrorCode NebulaStore::multiPutWithoutReplicator(GraphSpaceID spac
...
+  auto space = nebula::value(spaceRet);
+
+  folly::dynamic obj = folly::dynamic::object;
+  for (size_t i = 0; i < space->engines_.size(); i++) {
+    auto val = space->engines_[i]->getProperty(property);
+    if (!ok(val)) {
+      return error(val);
+    }
+    auto eng = folly::stringPrintf("Engine %zu", i);
+    obj[eng] = std::move(value(val));
```
Comment (bright-starry-sky) [inline src/kvstore/NebulaStore.cpp:1187]: nice

**Comment by darionyaphet** [comment]
only display on HTTP interface ?

**Comment by critical27** [comment]
> only display on HTTP interface ?

Yeah, rocksdb stats also provide in http only

**Review by bright-starry-sky — APPROVED** [review]
LGTM. do we submit this PR to 2.6?

**Comment by critical27** [comment]
> LGTM. do we submit this PR to 2.6?

Yeah, I have tested before. The compatibility need to check carefully later.

**Review by bright-starry-sky — APPROVED** [review]
LGTM

File: src/kvstore/RocksEngineConfig.cpp [hunk 2]
```diff
@@ -296,8 +264,9 @@ rocksdb::Status initRocksdbOptions(rocksdb::Options& baseOpts,
           baseOpts.compaction_style == rocksdb::CompactionStyle::kCompactionStyleLevel;
     }
     if (FLAGS_enable_rocksdb_prefix_filtering) {
-      baseOpts.prefix_extractor.reset(new GraphPrefixTransform(prefixLength));
+      baseOpts.prefix_extractor.reset(rocksdb::NewCappedPrefixTransform(prefixLength));
     }
+    bbtOpts.whole_key_filtering = FLAGS_enable_rocksdb_whole_key_filtering;
```
Comment (panda-sheep) [inline src/kvstore/RocksEngineConfig.cpp:269]: Have a question:
When `enable_rocksdb_prefix_filtering `and `enable_rocksdb_whole_key_filterin `are both true, 
which one is used?

**Review by panda-sheep — COMMENTED** [review]
Good job

File: src/kvstore/RocksEngineConfig.cpp (hunk 2 above)
Comment (critical27) [reply src/kvstore/RocksEngineConfig.cpp:269]: Both of them could used at the same time.

File: src/kvstore/RocksEngine.cpp [hunk 3]
```diff
@@ -213,7 +213,32 @@ nebula::cpp2::ErrorCode RocksEngine::range(const std::string& start,
...
+  }
+  storageIter->reset(new RocksPrefixIter(iter, prefix));
+  return nebula::cpp2::ErrorCode::SUCCEEDED;
+}
+
+nebula::cpp2::ErrorCode RocksEngine::prefixWithoutExtractor(
+    const std::string& prefix, std::unique_ptr<KVIterator>* storageIter) {
   rocksdb::ReadOptions options;
+  // prefix_same_as_start is false by default
+  options.total_order_seek = FLAGS_enable_rocksdb_prefix_filtering;
```
Comment (panda-sheep) [inline src/kvstore/RocksEngine.cpp:241]: 👍

**Review by panda-sheep — APPROVED** [review]

**Pull request merged**